chain/index/indexer.go

```diff
@@ -0,0 +1,416 @@
+package index
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/crypto"
+
+	"github.com/filecoin-project/lotus/chain/actors/builtin"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/lib/sqlite"
+)
+
+var _ Indexer = (*SqliteIndexer)(nil)
+
+// ActorToDelegatedAddressFunc is a function type that resolves an actor ID to a DelegatedAddress if one exists for that actor, otherwise returns nil
+type ActorToDelegatedAddressFunc func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) (address.Address, bool)
+type emsLoaderFunc func(ctx context.Context, cs ChainStore, msgTs, rctTs *types.TipSet) ([]executedMessage, error)
+type recomputeTipSetStateFunc func(ctx context.Context, ts *types.TipSet) error
```

Comment:
```suggestion
type RecomputeTipSetStateFunc func(ctx context.Context, ts *types.TipSet) error
```

since you're expecting it to be set publicly